api/server/backend/build/backend.go

```diff
@@ -25,10 +25,17 @@ type Backend struct {
 	imageComponent ImageComponent
 }
 
+type monolithBackend interface {
+	ImageComponent
+	builder.Backend
+	containerComponent
+}
+
 // NewBackend creates a new build backend from components
-func NewBackend(components ImageComponent, builderBackend builder.Backend) *Backend {
-	manager := dockerfile.NewBuildManager(builderBackend)
-	return &Backend{imageComponent: components, manager: manager}
+func NewBackend(monolith monolithBackend) *Backend {
+	containerBackendFunc := func() builder.ContainerBackend { return NewContainerBackend(monolith) }
+	manager := dockerfile.NewBuildManager(monolith, containerBackendFunc)
```

Comment:
Yes, this callback exists because each build should have it's own `containerBackend`.

> I don't think there is a need for these objects to be stateful

There is a need for **some** object to be stateful to manage temporary contaienrs. This object is current the `containerBackend`, so it must be stateful. 

>  It is dangerous to make such assumptions as 2 build jobs should be able to share each other's resources. 

I think the problem is that you asked me to move this out of the builder. Previously this made a lot more sense because it was part of the builder and not a dependency.